client/App.js

```diff
@@ -106,16 +100,45 @@ export default class App extends React.Component {
       'SF Pro Text Bold': require('./assets/fonts/SFProText/SF-Pro-Text-Bold.ttf'),
     })
 
+    await registerForPushNotificationsAsync();
+
+    // Push notifications listener <- the observer!
+    this.listener = Notifications.addListener(this.handleNotification);
+
     // App is ready to be loaded.
     this.setState({ appHasLoaded: true });
 
   }
 
+  // Callback to run when we observe a new
+  // push notification
+  handleNotification = (notification) => {
+    this.setState({ notification });
+  };
+
+  // Renders the notification component
+  // with the proper notification data
+  // If no notification data is passed in, render nothing.
+  NotificationContainer = () => {
+    if (!this.state.notification.data) return null;
+    const { title, message } = this.state.notification.data;
+    const dismiss = () => this.setState({notification: undefined});
+
+    return (
+      <Toast title={title || "Dining*v2"} message={message || "New alert."} onPress={dismiss} />
+    )
+  }
+
+
   render() {
+    const { NotificationContainer } = this;
```

Comment:
This is just using Javascript's [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring) to try to make the code a bit cleaner. Since `NotificationContainer` lives in the `this` scope, I would technically have to use the component in JSX form as: `<this.NotificationContainer />` but instead I declared a local variable called `NotificationContainer` that I pulled out of the `this` context, if that makes sense.

Reply:
Ah! For some reason I was thinking the assignment went the other way...I belong in 201...